### PR TITLE
add opensslextra and opensslall configure options for wolfssl

### PR DESCRIPTION
**Pull request by kirandivekar**
Add support for opensslextra, opensslall configuration options for openssl compatibility layer.
With opensslextra only partial functionality gets enabled. In order to enable complete functionality,
using all OpenSSL APIs, opensslall flag is needed.
Default value for both options is false.

Specify library name and version:  **wolfssl/4.4.0**

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by kirandivekar** [comment]
Based on PR 2167

File: recipes/wolfssl/all/conanfile.py [hunk 1]
```diff
@@ -76,6 +80,10 @@ def _configure_autotools(self):
             "--enable-harden",
             "--enable-debug" if self.settings.build_type == "Debug" else "--disable-debug",
         ]
+        if self.options.opensslextra:
+            conf_args.extend(["--enable-opensslextra"])
+            if self.options.opensslall:
+                conf_args.extend(["--enable-opensslall"])
```
Comment (madebr) [inline recipes/wolfssl/all/conanfile.py:86]: Does this make sense?
```suggestion
            "--enable-opensslextra" if self.options.opensslextra else "--disable-opensslextra",
            "--enable-opensslall" if self.options.opensslall else "--disable-opensslall",
        ]
```

**Comment by kirandivekar** [comment]
Will take care of the comment to remove `self.options.opensslall` when `self.options.opensslextra` is false.

**Comment by kirandivekar** [comment]
I re-thought about your suggestion. The `openssl` options are not environment specific or generic from perspective of wolfssl. One who needs openssl will use either of `opensslextra` OR `opensslextra + opensslall` OR `opensslall` only.

I checked again that only having `opensslall` sliently enables "opensslextra as well. I will change my patch to reflect this behaviour. Let me know, if there is any suggestion.

Thanks for the review.

**Comment by kirandivekar** [comment]
This is moved to PR 2294

**Pull request closed without merging**